packages/cli/src/commands/deploy.ts

```diff
@@ -40,73 +40,239 @@ import {
+      deprecated: false,
+      description: 'TODO: Description for --prebuilt',
+      multi: false,
+    },
+    {
+      name: 'prod',
+      shorthand: null,
+      type: 'boolean',
+      deprecated: false,
+      description: 'Create a production deployment',
+      multi: false,
+    },
+    {
+      name: 'archive',
+      shorthand: null,
+      type: 'string',
+      deprecated: false,
+      description: 'TODO: Description for --archive',
+      multi: false,
+    },
+    {
+      name: 'no-wait',
+      shorthand: null,
+      type: 'boolean',
+      deprecated: false,
+      description: "Don't wait for the deployment to finish",
+      multi: false,
+    },
+    {
+      name: 'skip-domain',
+      shorthand: null,
+      type: 'boolean',
+      deprecated: false,
+      description: 'TODO: Description for --skip-domain',
+      multi: false,
+    },
+    {
+      name: 'yes',
+      shorthand: 'y',
+      type: 'boolean',
+      deprecated: false,
+      description: 'TODO: Description for --yes',
+      multi: false,
+    },
+    {
+      name: 'name',
+      shorthand: 'n',
+      type: 'string',
+      deprecated: true,
+      description: 'TODO: Description for --name',
+      multi: false,
+    },
+    {
+      name: 'no-clipboard',
+      shorthand: null,
+      type: 'boolean',
+      deprecated: true,
+      description: 'TODO: Description for --no-clipboard',
+      multi: false,
+    },
+    {
+      name: 'target',
+      shorthand: null,
+      type: 'string',
+      deprecated: true,
+      description: 'TODO: Description for --target',
+      multi: false,
+    },
+    {
+      name: 'confirm',
+      shorthand: 'c',
+      type: 'boolean',
+      deprecated: true,
+      description: 'TODO: Description for --confirm',
+      multi: false,
+    },
+  ],
+  examples: [
+    {
+      name: 'Deploy the current directory',
+      value: 'vercel',
+    },
+    {
+      name: 'Deploy a custom path',
+      value: 'vercel /usr/src/project',
+    },
+    {
+      name: 'Deploy with run-time Environment Variables',
+      value: 'vercel -e NODE_ENV=production',
+    },
+    {
+      name: 'Deploy with prebuilt outputs',
+      value: ['vercel build', 'vercel deploy --prebuilt'],
+    },
+    {
+      name: 'Write Deployment URL to a file',
+      value: 'vercel > deployment-url.txt',
+    },
+  ],
+};
 
 export default async (client: Client): Promise<number> => {
   const { output } = client;
 
   let argv = null;
 
+  const argOptions: {
+    [k in string]:
+      | BooleanConstructor
+      | StringConstructor
+      | string
+      | [StringConstructor];
+  } = {};
+  for (const option of deployCommand.options) {
+    argOptions[`--${option.name}`] =
+      option.type === 'boolean' ? Boolean : String;
+    if (option.shorthand) {
+      argOptions[`-${option.shorthand}`] = `--${option.name}`;
+    }
+    if (
+      option.name === 'env' ||
+      option.name === 'build-env' ||
+      option.name === 'meta'
```

Comment:
<!-- Questions comments seek clarification.  -->
**question (non-blocking):** ‏Can we encode this in the Command definition instead of special casing it here?

Reply:
I didn't see a clear way to do so in the definition. Our arg option handling could be improved whole-sale too at a later point. But I want the new stuff to work as is with the existing args handling.